kernel/src/main/java/org/kframework/kompile/KompileOptions.java

```diff
@@ -98,42 +98,36 @@ public boolean strict() {
+    @Parameter(names="--heat-cool-by-strategies", description="Control heating and cooling using strategies.")
+    public boolean heatCoolStrategies = false;
 
-        @ParametersDelegate
-        public SMTOptions smt = new SMTOptions();
+    @ParametersDelegate
+    public SMTOptions smt = new SMTOptions();
 
-        @Parameter(names="--cache-file", description="Location of parse cache file. Default is $KOMPILED_DIR/cache.bin.")
-        public String cacheFile;
+    @Parameter(names="--cache-file", description="Location of parse cache file. Default is $KOMPILED_DIR/cache.bin.")
+    public String cacheFile;
 
-        @Parameter(names="--emit-json", description="Emit JSON serialized version of parsed and kompiled definitions.")
-        public boolean emitJson = false;
+    @Parameter(names="--emit-json", description="Emit JSON serialized version of parsed and kompiled definitions.")
+    public boolean emitJson = false;
 
-        @Parameter(names="--gen-bison-parser", description="Emit bison parser for the PGM configuration variable within the syntax module of your definition into the kompiled definition.")
-        public boolean genBisonParser;
+    @Parameter(names="--gen-bison-parser", description="Emit bison parser for the PGM configuration variable within the syntax module of your definition into the kompiled definition.")
+    public boolean genBisonParser;
 
-        @Parameter(names="--gen-glr-bison-parser", description="Emit GLR bison parser for the PGM configuration variable within the syntax module of your definition into the kompiled definition.")
-        public boolean genGlrBisonParser;
+    @Parameter(names="--gen-glr-bison-parser", description="Emit GLR bison parser for the PGM configuration variable within the syntax module of your definition into the kompiled definition.")
+    public boolean genGlrBisonParser;
 
-        @Parameter(names="--bison-file", description="C file containing functions to link into bison parser.")
-        public String bisonFile;
+    @Parameter(names="--bison-file", description="C file containing functions to link into bison parser.")
+    public String bisonFile;
 
-        @Parameter(names="--bison-stack-max-depth", description="Maximum size of bison parsing stack (default: 10000).")
-        public long bisonStackMaxDepth = 10000;
+    @Parameter(names="--bison-stack-max-depth", description="Maximum size of bison parsing stack (default: 10000).")
+    public long bisonStackMaxDepth = 10000;
 
-        @Parameter(names="--transition", listConverter=StringListConverter.class, description="<string> is a whitespace-separated list of tags designating rules to become transitions.")
-        public List<String> transition = Collections.singletonList(DEFAULT_TRANSITION);
+    @Parameter(names="--transition", listConverter=StringListConverter.class, description="<string> is a whitespace-separated list of tags designating rules to become transitions.")
```

Comment:
We should probably at the very least mark this option as deprecated because I don't believe it has any meaning on the Haskell backend.

Reply:
I would like to remove this option altogether in a separate PR, because I remember that we discovered that it's actually basically hopelessly broken in the past, and had to rely on other methods to achieve the same thing (like just using the strategy module directly).

So maybe I can make an issue about removing it and handle it in another PR? The main argument for that being that it will require some updates to the PL-tutorial, which should be reviewed separately.

Reply:
See #1936

Reply:
I've marked this as deprecated/java backend only.